### PR TITLE
Add tutorial for 3D perspective image

### PR DESCRIPTION
**Pull request by willschlitzer**
Tutorial to create an Earth-relief three-dimensional figure.

### CODE REVIEW

File: examples/tutorials/3d-map.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+"""
+Creating a 3D map
+=================
+
+Plotting a three-dimensional map is handled by :meth:`pygmt.Figure.grdview`.
```
Comment (seisman) [inline examples/tutorials/3d-map.py:5]: This is not correct. Most plotting modules can plot 3D maps, for example https://www.pygmt.org/dev/gallery/plot/scatter3d.html.

Perhaps we can say:

> Create a 3-D perspective image

File: examples/tutorials/3d-map.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+# default is [180, 90], which is looking directly down on the figure and north is "up".
+# The ``zsize`` argument sets how tall the three-dimensional portion appears.
+#
+# The default figure surface is *mesh plot*.
+
+fig = pygmt.Figure()
+fig.grdview(
+    grid=grid,
+    # Sets the view azimuth as 180 degrees, and the view angle as 30 degrees
+    perspective=[180, 30],
```
Comment (seisman) [inline examples/tutorials/3d-map.py:25]: I understand why you choose az=180 here, but the figure doesn't look good to me. What about using az=130 here, and then change it to az=-130 or other values in the later section?

File: examples/tutorials/3d-map.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+# Load sample earth relief data
+grid = pygmt.datasets.load_earth_relief(resolution="05m", region=[-108, -103, 35, 40])
+
+########################################################################################
+# The :meth:`pygmt.Figure.grdview` method takes the ``grid`` input.
+# The ``perspective`` argument changes the azimuth and angle of the viewpoint; the
+# default is [180, 90], which is looking directly down on the figure and north is "up".
+# The ``zsize`` argument sets how tall the three-dimensional portion appears.
+#
+# The default figure surface is *mesh plot*.
```
Comment (seisman) [inline examples/tutorials/3d-map.py:19]: This sentence doesn't read well to me. Perhaps change to "The default grid surface type is *mesh plot*"?

File: examples/tutorials/3d-map.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+# The default CPT is *turbo*.
+
+fig = pygmt.Figure()
+fig.grdview(
+    grid=grid,
+    perspective=[180, 30],
+    frame=["xa", "ya", "WSnE"],
+    projection="M15c",
+    zsize="1.5c",
+    # Sets the surface type to solid
```
Comment (seisman) [inline examples/tutorials/3d-map.py:46]: I think **s** means *surface* here. See https://docs.generic-mapping-tools.org/latest/grdview.html#q

> Specify s for surface plot, and optionally append m to have mesh lines drawn on top of surface.

File: examples/tutorials/3d-map.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+fig = pygmt.Figure()
+fig.grdview(
+    grid=grid,
+    perspective=[180, 30],
+    frame=["xa", "yaf", "WSnE"],
+    projection="M15c",
+    zsize="1.5c",
+    surftype="s",
+    cmap="geo",
```
Comment (seisman) [inline examples/tutorials/3d-map.py:79]: Please add a comment here to explain what `1000+ggrey` does.

File: examples/tutorials/3d-map.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+fig.show()
+
+########################################################################################
+# The ``perspective`` azimuth can be changed to set the direction that is "up"
+# in the figure.
+
+fig = pygmt.Figure()
+fig.grdview(
+    grid=grid,
+    # Set the azimuth to 135 degrees and the angle to 30 degrees
```
Comment (seisman) [inline examples/tutorials/3d-map.py:91]:
```suggestion
    # Set the azimuth to 135 degrees and the elevation to 30 degrees
```

File: examples/tutorials/3d-map.py (hunk 1 above)
Comment (willschlitzer) [reply examples/tutorials/3d-map.py:5]: Changed.

File: examples/tutorials/3d-map.py (hunk 2 above)
Comment (willschlitzer) [reply examples/tutorials/3d-map.py:25]: Changed to your suggestion.

File: examples/tutorials/3d-map.py (hunk 3 above)
Comment (willschlitzer) [reply examples/tutorials/3d-map.py:19]: Changed to your suggestion.

File: examples/tutorials/3d-map.py (hunk 4 above)
Comment (willschlitzer) [reply examples/tutorials/3d-map.py:46]: I even looked that up as I was writing the comment and still managed to type it incorrectly. Changed,

File: examples/tutorials/3d-map.py (hunk 5 above)
Comment (willschlitzer) [reply examples/tutorials/3d-map.py:79]: Added.

File: doc/index.rst [hunk 7]
```diff
@@ -36,6 +36,7 @@
     tutorials/text.rst
     tutorials/contour-map.rst
     tutorials/earth-relief.rst
+    tutorials/3d-map.rst
```
[the remark this answers is not in the document]
Comment (willschlitzer) [reply doc/index.rst:39]: Renamed to "3d-perspective-image"

File: examples/tutorials/3d-map.py [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    plane="1000+ggrey",
+    # Set the contour pen thickness to "0.5p"
+    contourpen="0.5p",
+)
+fig.show()
+
+########################################################################################
+# :meth:`pygmt.Figure.colorbar` can be used to add a color bar to the figure. The
+# ``cmap`` argument does not need to be passed again. To keep the color bar's alignment
+# similar to the figure, it also takes a ``perspective`` argument.
```
[the remark this answers is not in the document]
Comment (willschlitzer) [reply examples/tutorials/3d-map.py]: Done.

**Review by seisman — APPROVED** [review]

**Pull request merged**